internal/discovery/host.go

```diff
@@ -50,15 +52,21 @@ func (d HostDiscovery) Discover(ctx context.Context) (string, error) {
 		return "", err
 	}
 
+	var discoveredHostFQDN *string
```

Comment:
do we need this extra local variable? `d.fqdn` might come nullish (empty string) anyway from the caller, right? Or am I missing something?